pyqldb/session/qldb_session.py

```diff
@@ -100,128 +91,66 @@ def session_token(self):
         """
         return self._session.token
 
-    def _close(self):
-        """
-        Close this `QldbSession`.
-        """
-        self._is_closed = True
-
     def _end_session(self):
         """
         End this session. No-op if already closed.
         """
-        if not self._is_closed:
-            self._is_closed = True
+        if self._is_alive:
+            self._is_alive = False
```

Comment:
Not unless the user calls the private functions directly.

```
qldb_session = qldb_driver._get_session(True)
qldb_session._close()
qldb_session._execute_lambda(...)
```

But yeah you are right, since the driver class is responsible for session management, it won't put a dead session back into pool.